src/components/generic/ContentVersionControls.tsx

```diff
@@ -233,6 +245,49 @@ class ContentVersionControls extends React.Component<Props, State> {
     );
   }
 
+  /**
+   * Renders delete version button
+   */
+  private renderDeleteVersionButton = () => {
+    const { classes } = this.props;
+
+    return (
+      <Tooltip title={ strings.contentVersionControls.deleteSelectedVersion }>
+        <Button
+          disableElevation
+          className={ classes.deleteButton }
+          color="primary"
+          variant="contained"
+          onClick={ () => this.setState({ deleteVersionDialogOpen: true }) }
+        >
+          { strings.contentVersionControls.deleteVersion }
+        </Button>
+      </Tooltip>
+    );
+  }
+
+  /**
+   * Render delete version confirmation dialog
+   */
+  private renderDeleteVersionDialog = () => {
+    const { deleteVersionDialogOpen } = this.state;
+
+    return (
+      <GenericDialog
+        title={ strings.contentVersionControls.deleteVersion }
+        onCancel={ () => this.setState({ deleteVersionDialogOpen: false }) }
+        onClose={ () => this.setState({ deleteVersionDialogOpen: false }) }
+        onConfirm={ () => this.onDeleteVersionClick() }
```

Comment:
Why this method is invoked here?
```suggestion
        onConfirm={ this.onDeleteVersionClick }
```